main.go

```diff
@@ -95,37 +96,36 @@ func main() {
 	}
 }
 
-func run(ctx context.Context, r *resources.GroupsClient, ttl time.Duration, dryRun bool) error {
+func run(ctx context.Context, r *armresources.ResourceGroupsClient, ttl time.Duration, dryRun bool) error {
 	log.Println("Scanning for stale resource groups")
-	listComplete, err := r.ListComplete(ctx, "", nil)
-	if err != nil {
-		return fmt.Errorf("Error when listing all resource groups: %v", err)
-	}
-	for list := listComplete; list.NotDone(); err = list.Next() {
+
+	pager := r.NewListPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
 		if err != nil {
 			return fmt.Errorf("Error when iterating resource groups: %v", err)
 		}
-
-		rg := list.Value()
-		rgName := *rg.Name
-		if age, ok := shouldDeleteResourceGroup(rg, ttl); ok {
-			// Does not wait when deleting a resource group
-			log.Printf("Deleting resource group '%s' (age: %s)", rgName, age)
-			if dryRun {
-				continue
-			}
-
-			_, err = r.Delete(ctx, rgName)
-			if err != nil {
-				log.Printf("Error when deleting %s: %v", rgName, err)
+		for _, rg := range nextResult.Value {
+			rgName := *rg.Name
+			if age, ok := shouldDeleteResourceGroup(rg, ttl); ok {
+				// Does not wait when deleting a resource group
```

Comment:
Not sure what this comment means

Reply:
i think it means we're just doing a fire-and-forget operation. We don't wait for the deletions to finish or check the status

Reply:
clarified comment